sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/BoundAttribute.scala

```diff
@@ -23,6 +23,15 @@ import org.apache.spark.sql.catalyst.errors.attachTree
 import org.apache.spark.sql.catalyst.expressions.codegen.{CodeGenContext, GeneratedExpressionCode}
 import org.apache.spark.sql.types._
 
+object BoundReference {
+  def apply(
```

Comment:
oh, this a old pr. :-)

This change takes cares of the corner case that a `BoundReference` is aliased to a name and has been removed of the `Alias` later in codegen. I think not all `BoundReference` have a name. So I make it as optional field. Not sure about whether the case you said will happen.